### PR TITLE
Clean up Spec Test dependencies

### PR DESCRIPTION
**Pull request by mikaelm12**
Updated the dependencies of the Specification Tests package. 
 - Removed AspNetCore.Testing
 - Used the Xunit extensibility packages

This still depends on Moq because of  `HubConnectionContextUtils`

### CODE REVIEW

File: src/Microsoft.AspNetCore.SignalR.Specification.Tests/Microsoft.AspNetCore.SignalR.Specification.Tests.csproj [hunk 1]
```diff
@@ -28,11 +28,10 @@
 
   <ItemGroup>
     <PackageReference Include="Internal.AspNetCore.Sdk" PrivateAssets="All" Version="$(InternalAspNetCoreSdkPackageVersion)" />
-    <PackageReference Include="Microsoft.AspNetCore.Testing" Version="$(MicrosoftAspNetCoreTestingPackageVersion)" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="$(MicrosoftNETTestSdkPackageVersion)" />
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.SignalR.Specification.Tests/Microsoft.AspNetCore.SignalR.Specification.Tests.csproj]: Can you remove this too?

File: src/Microsoft.AspNetCore.SignalR.Specification.Tests/Microsoft.AspNetCore.SignalR.Specification.Tests.csproj [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <Description>Tests for users to verify their own implementations of SignalR types</Description>
     <!-- Temporarily suppress packing this project until we can fix issues in its dependencies. -->
     <IsPackable>false</IsPackable>
-    <DeveloperBuildTestTfms>netcoreapp2.2</DeveloperBuildTestTfms>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
     <StandardTestTfms>$(DeveloperBuildTestTfms)</StandardTestTfms>
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.SignalR.Specification.Tests/Microsoft.AspNetCore.SignalR.Specification.Tests.csproj]: I think you can remove the next 6 lines as well as IsPackable=false (above).

Also, just to be clear, we're explicitly not supporting netstandard2.0 for these tests, right?

File: src/Microsoft.AspNetCore.SignalR.Specification.Tests/Microsoft.AspNetCore.SignalR.Specification.Tests.csproj [hunk 3]
```diff
@@ -28,11 +28,10 @@
 
   <ItemGroup>
     <PackageReference Include="Internal.AspNetCore.Sdk" PrivateAssets="All" Version="$(InternalAspNetCoreSdkPackageVersion)" />
-    <PackageReference Include="Microsoft.AspNetCore.Testing" Version="$(MicrosoftAspNetCoreTestingPackageVersion)" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="$(MicrosoftNETTestSdkPackageVersion)" />
     <PackageReference Include="Moq" Version="$(MoqPackageVersion)" />
-    <PackageReference Include="xunit" Version="$(XunitPackageVersion)" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="$(XunitRunnerVisualStudioPackageVersion)" />
+    <PackageReference Include="xunit.extensibility.core" Version="$(XunitPackageVersion)" />
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.SignalR.Specification.Tests/Microsoft.AspNetCore.SignalR.Specification.Tests.csproj]: Nit: `Version="$(XunitExtensibilityCorePackageVersion)"` and `Version="$(XunitAssertPackageVersion)"`

File: src/Microsoft.AspNetCore.SignalR.Specification.Tests/Microsoft.AspNetCore.SignalR.Specification.Tests.csproj [hunk 4]
```diff
@@ -28,11 +28,10 @@
 
   <ItemGroup>
     <PackageReference Include="Internal.AspNetCore.Sdk" PrivateAssets="All" Version="$(InternalAspNetCoreSdkPackageVersion)" />
-    <PackageReference Include="Microsoft.AspNetCore.Testing" Version="$(MicrosoftAspNetCoreTestingPackageVersion)" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="$(MicrosoftNETTestSdkPackageVersion)" />
     <PackageReference Include="Moq" Version="$(MoqPackageVersion)" />
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.SignalR.Specification.Tests/Microsoft.AspNetCore.SignalR.Specification.Tests.csproj]: Would be preferable to internalize this dependency if possible.

**Comment by mikaelm12** [comment]
Completely removed our dependency on Moq in this package with a custom implementations of HubConnectionContext that does what we need for testing

File: test/Microsoft.AspNetCore.SignalR.Tests.Utils/HubConnectionContextUtils.cs [hunk 5]
```diff
@@ -25,13 +27,24 @@ public static HubConnectionContext Create(ConnectionContext connection, IHubProt
...
+        public static MockHubConnectionContext CreateMock(ConnectionContext connection)
         {
-            var mock = new Mock<HubConnectionContext>(connection, TimeSpan.FromSeconds(15), NullLoggerFactory.Instance, TimeSpan.FromSeconds(15)) { CallBase = true };
-            var protocol = new JsonHubProtocol();
-            mock.SetupGet(m => m.Protocol).Returns(protocol);
-            return mock;
+            return new MockHubConnectionContext(connection, TimeSpan.FromSeconds(15), NullLoggerFactory.Instance, TimeSpan.FromSeconds(15));
+        }
+
+        public class MockHubConnectionContext : HubConnectionContext
```
Comment (anurse) [inline test/Microsoft.AspNetCore.SignalR.Tests.Utils/HubConnectionContextUtils.cs]: 👍

**Pull request merged**